src/Control/Monad/Trace.hs

```diff
@@ -50,6 +59,7 @@ import qualified Data.Map.Strict as Map
 import Data.Maybe (fromMaybe)
 import Data.Time.Clock (NominalDiffTime)
 import Data.Time.Clock.POSIX (POSIXTime, getPOSIXTime)
+import GHC.Conc (retry)
```

Comment:
While we're at it, can you get rid of the `stm-lifted` dependency? It's abandoned and not really needed here, AFAIK the only thing it gives us is lifted `atomically`.

Then you can use `import Control.Concurrent.STM` and it imports `retry`.

Reply:
Nevermind, let's do it in a different PR.

You can use `Control.Monad.STM` though instead of `GHC.Conc`.

Reply:
I can, but (a) it's already a reexport for GHC.Conc (b) it requires adding the stm dependency (which is admittedly only hidden for now). I will gladly do a PR dropping stm-lifted as a follow-up to this one.

Reply:
Ok, let's leave it then.

Reply:
Yes, I was explaining why I had used GHC.Conc, not refusing the requested change ! If the state of the PR seems satisfactory to you, I think we can merge and I'll do the PR dropping stm-lifted right after.

Reply:
Sounds good :+1: